edit contacts with the help of dialog box #17

MayankUjawane opened the pull request:
• Added the edit feature.
• Dialog box will appear with the previous name and number.
• Added the checks for the details provided by the user.

codeDamon commented on app/src/main/java/com/example/she/ViewContacts.java:
```diff
@@ -25,25 +33,86 @@
+        builder.setView(viewGroup)
+                .setTitle("Edit Contact Details")
+                .setIcon(R.drawable.ic_edit_24)
+                .setPositiveButton("Update", null)
```
Instead of passing null in line 59, you can make these changes: 
* Delete lines 67-78
* Create a function 'updateContact()' and add relevant code of these lines(67-78) inside it.
* Now in line 59, instead of null, create DialogInterface.OnClickListener() similar to done in line 60.
* Now call updateContact() inside onClick()

Hope you understant that!

MayankUjawane replied:
Sorry but I am not able to understand the benefit of this.

MayankUjawane replied:
Hey @codeDamon I have tried your suggestion but then the code is not working as wanted.

codeDamon replied:
> Hey @codeDamon I have tried your suggestion but then the code is not working as wanted.

I doubt why it would not work. 
Unzip attached file. It is the edited ViewContacts.java.
I tested it, its working fine.
Also one more thing, remove the pink background color from edit_contact.xml, it look unprofessional.

[ViewContacts.java.zip](https://github.com/priyalbhatewara123/SHE--Live-Without-Fear/files/5977850/ViewContacts.java.zip)

codeDamon reviewed (changes requested):
Good Job, @MayankUjawane 
There is a minor improvement I feel we need. Check it out.
Do confirm with @priyalbhatewara123 before changing.

priyalbhatewara123 reviewed (commented):
Few things to be updated:
1. UI of dialog box doesn't look good. 
    * Remove the pink background. 
    * There's no need for the title ("Edit Contact Details").
    * Add textview above edittext which says Name and Number.
    * You may use material design for more creativity.
 2. If I update only name then it says contact already exist. So add a functionality to edit only the name.
 3. Add a toast "Contact Updated" after the user edit the contact.

MayankUjawane commented:
Updated the edit functionality according to your suggestion.
• Added the material design in UI.
• Removed the pink background of Dialog box.
• Added Toast.
• Solved the issue of update name.

priyalbhatewara123 commented:
> Updated the edit functionality according to your suggestion.
> • Added the material design in UI.
> • Removed the pink background of Dialog box.
> • Added Toast.
> • Solved the issue of update name.

Can you please send a screenshot of the dialog box. @MayankUjawane

MayankUjawane commented:
![Screenshot (55)](https://user-images.githubusercontent.com/67855648/108238413-cb9c7f80-70fd-11eb-8c17-9c51f90d786f.png)

MayankUjawane commented:
Please review Ma'am

priyalbhatewara123 commented:
> Please review Ma'am

This looks nice. Can you please send me an apk file so that I can test on my phone.

MayankUjawane commented:
[app-debug.zip](https://github.com/priyalbhatewara123/SHE--Live-Without-Fear/files/5997626/app-debug.zip)
@priyalbhatewara123 Please check this apk

priyalbhatewara123 commented:
Great work! @MayankUjawane ✨
